### PR TITLE
[stable/jenkins] Allow templating of all string values except type for kubernetes volumes plugin xml

### PR DESCRIPTION
**Pull request by krishnakvall**
#### What this PR does / why we need it:

Allow templating of all string values except type for volumes list in kubernetes volumes plugin xml

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Special notes for your reviewer:

These values are already templated for the jenkins.casc.podTemplate helper function. This PR repeats the same templating for the jenkins.xml.podTemplate helper function. 

This will allow us to gererate chart specfic names for volume mounts and configure the kubernetes plugin accordingly with xml. 

Code Snippets to illustrate the functionality that has been duplicated.

Templating already present in:
```
{{- define "jenkins.casc.podTemplate" -}}
...
        {{ $key }}: {{ if kindIs "string" $value }}{{ tpl $value $ | quote }}{{ else }}{{ $value }}{{ end }}
```

Templating introduced for:
```
{{- define "jenkins.xml.podTemplate" -}}
...
      <{{ $key }}>{{ if kindIs "string" $value }}{{ tpl $value $ | quote }}{{ else }}{{ $value }}{{ end -}}</{{ $key }}>
```

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by torstenwalter** [comment]
looks good to me @wmcdona89 What do you think>
@krishnakvall Can you fix the merge conflict?

**Comment by krishnakvall** [comment]
Merge conflicts have been fixed @torstenwalter

File: stable/jenkins/templates/_helpers.tpl [hunk 1]
```diff
@@ -275,7 +275,7 @@ Returns kubernetes pod template xml configuration
     <org.csanchez.jenkins.plugins.kubernetes.volumes.{{ $volume.type }}Volume>
   {{- end }}
   {{- range $key, $value := $volume }}{{- if not (eq $key "type") }}
-      <{{ $key }}>{{ $value }}</{{ $key }}>
+      <{{ $key }}>{{ if kindIs "string" $value }}{{ tpl $value $ | quote }}{{ else }}{{ $value }}{{ end -}}</{{ $key }}>
```
Comment (wmcdona89) [inline stable/jenkins/templates/_helpers.tpl]: I don't think the value needs to be quoted in XML. Use of `quote` results in `<mountPath>"/var/myapp/myemptydir"</mountPath>`

Comment (wmcdona89) [inline stable/jenkins/templates/_helpers.tpl]: the chomp in `{{ end -}}` is extraneous and can be removed

**Review by wmcdona89 — CHANGES_REQUESTED** [review]
@krishnakvall thanks for the PR!

File: stable/jenkins/templates/_helpers.tpl (hunk 1 above)
Comment (krishnakvall) [reply stable/jenkins/templates/_helpers.tpl]: Thank you @wmcdona89 . I have merged the latest code back and addressed these comments. 

Note: 
1. the latest change 1.23.1 did not seem to have a CHANGELOG entry when I merged
2. Had to rebase to sign an older commit

Comment (wmcdona89) [reply stable/jenkins/templates/_helpers.tpl]: @krishnakvall the updates look good. I see the 1.23.0 CHANGELOG entry is now missing. can you please restore it and also add a 1.23.1 CHANGELOG entry. thanks!

```
1.23.1

Added auto detection for Ingress API version
```

Comment (krishnakvall) [reply stable/jenkins/templates/_helpers.tpl]: Hi @wmcdona89 ,  have added 1.23.1 and fixed the removal of 1.23.0. Thank you!

**Comment by wmcdona89** [comment]
@krishnakvall thanks again for the contribution!

**Pull request merged**